### PR TITLE
Feat: UI for listing extension requests in /requests page

### LINKED ISSUE #756: Create UI for Listing Extension Requests in /requests Page
### Issue Description
Develop a user interface to list extension requests separately from OOO requests within the /requests page. Super users should be able to update the status of extension requests (e.g., approve, reject) directly from this page.

### Expected Behavior
The UI should include:
- A separate tab for listing extension requests.
- Controls for super users to update the status of extension requests (approve, reject).

### Current Behavior
The /request page already lists OOO requests but lacks a separate tab for extension requests and controls for super users.

### Reproducibility
- [ ] This issue is reproducible
- [x] This issue is not reproducible

### Steps to Reproduce
N/A (UI creation task)

### Severity/Priority
- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by Achintya-Chatterjee**
Date: 6th August 2024

Developer Name: @Achintya-Chatterjee 

---

## Issue Ticket Number

- Closes #756
## Description

- Integrated the extension request section in the /request page on the extension tab

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

https://github.com/user-attachments/assets/7b7a3dde-75fe-4aa9-b7ad-2023eab625e9

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-08-09 at 02 28 07

</details>

## Additional Notes
- I will add the test PR in the next PR because this PR is already big, meantime you can review this PR. #778

### CODE REVIEW

**Comment by sahsisunny** [comment]
Based on the demo video you provided:

- The loader should be centered in the display, and even if it’s specific to the ER card, it should still be positioned in the center.
- The query parameter is not showing when you switch between tabs, such as from OOO to Extension.

File: requests/util.js [hunk 1]
```diff
@@ -43,6 +43,154 @@ function getOooQueryParamsString(query) {
...
+async function updateExtensionRequestStatus({ id, body }) {
+  const url = `${API_BASE_URL}/extension-requests/${id}/status`;
+  const res = await fetch(url, {
+    credentials: 'include',
+    method: 'PATCH',
+    body: JSON.stringify(body),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
```
Comment (sahsisunny) [inline requests/util.js:79]: We have the `/requests` API available for all types of requests. Please use this API and refer to the API contract documentation for any related information.

File: requests/util.js [hunk 2]
```diff
@@ -43,6 +43,154 @@ function getOooQueryParamsString(query) {
...
+    ) {
+      params.append(key, query[key]);
+    }
+  }
+  return `?${params.toString()}`;
+}
+const Order = {
+  DESCENDING: 'desc',
+  ASCENDING: 'asc',
+};
```
Comment (sahsisunny) [inline requests/util.js:112]: The `/requests` route does not include sorting options.

File: requests/util.js [hunk 3]
```diff
@@ -43,6 +43,154 @@ function getOooQueryParamsString(query) {
...
+}
+async function getExtensionRequestLogs({ extensionRequestId }) {
+  const url = `${API_BASE_URL}/logs/extensionRequests/?meta.extensionRequestId=${extensionRequestId}`;
+  const res = await fetch(url, {
+    credentials: 'include',
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
```
Comment (sahsisunny) [inline requests/util.js:186]: We also need to update this endpoint if we're using the `/requests` route.

**Review by sahsisunny — CHANGES_REQUESTED** [review]

**Comment by sahsisunny** [comment]
### Here is the key difference between [old ](https://api.realdevsquad.com/extension-requests?order=asc&size=1)and [new ](https://staging-api.realdevsquad.com/requests?dev=true&type=EXTENSION)data
| **Attribute**       | **New**        | **Old**        |
|---------------------|-----------------------------|-------------------------------|
| **id**              | Present                      | Present                        |
| **createdAt**       | Present                      | Not Present                    |
| **requestedBy**     | Present                      | Not Present                    |
| **requestNumber**   | Present                      | Present                        |
| **newEndsOn**       | Present                      | Present                        |
| **oldEndsOn**       | Present                      | Present                        |
| **assignee**        | Present                      | Present                        |
| **type**            | Present                      | Not Present                    |
| **title**           | Present                      | Present                        |
| **message**         | Present                      | Not Present                    |
| **taskId**          | Present                      | Present                        |
| **lastModifiedBy**  | Present                      | Not Present                    |
| **state/status**    | Present (state)              | Present (status)               |
| **updatedAt**       | Present                      | Not Present                    |
| **reason**          | Not Present(same as `message`)                 | Present                        |
| **assigneeId**      | Not Present(same as `requestedBy`)                  | Present                        |
| **timestamp**       | Not Present(same as `createdBy` and `updatedat`)                  | Present                        |

**Comment by Achintya-Chatterjee** [comment]
> Based on the demo video you provided:
> 
> * The loader should be centered in the display, and even if it’s specific to the ER card, it should still be positioned in the center.
> * The query parameter is not showing when you switch between tabs, such as from OOO to Extension.
> * The filters and search options are the same across all tabs, such as OOO, Task, and Extension, and are not specific to the Extension tab only.

@sahsisunny I didn't develop this UI, it was already been built by devanshu, I just raise the PR

**Comment by sahsisunny** [comment]
> > Based on the demo video you provided:
> > 
> > * The loader should be centered in the display, and even if it’s specific to the ER card, it should still be positioned in the center.
> > * The query parameter is not showing when you switch between tabs, such as from OOO to Extension.
> > * The filters and search options are the same across all tabs, such as OOO, Task, and Extension, and are not specific to the Extension tab only.
> 
> @sahsisunny I didn't develop this UI, it was already been built by devanshu, I just raise the PR

I know @Achintya-Chatterjee . Your task is not only to be raise PR. You need also to do changes if required.

**Comment by Achintya-Chatterjee** [comment]
> > > Based on the demo video you provided:
> > > 
> > > * The loader should be centered in the display, and even if it’s specific to the ER card, it should still be positioned in the center.
> > > * The query parameter is not showing when you switch between tabs, such as from OOO to Extension.
> > > * The filters and search options are the same across all tabs, such as OOO, Task, and Extension, and are not specific to the Extension tab only.
> > 
> > 
> > @sahsisunny I didn't develop this UI, it has already been built by Devanshu, I just raised the PR
> 
> I know @Achintya-Chatterjee . Your task is not only to raise PR. You also need to do changes if required.

@sahsisunny then how do you want the new extension request to be?

**Comment by Achintya-Chatterjee** [comment]
> Based on the demo video you provided:
> 
> * The loader should be centered in the display, and even if it’s specific to the ER card, it should still be positioned in the center.
> * The query parameter is not showing when you switch between tabs, such as from OOO to Extension.
> * The filters and search options are the same across all tabs, such as OOO, Task, and Extension, and are not specific to the Extension tab only.

@sahsisunny there is no filter in OOO and Task.

**Review by ashifkhn — DISMISSED** [review]

**Review by ashifkhn — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**